firebase-firestore/src/main/java/com/google/firebase/firestore/AggregateField.java

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.RestrictTo;
+import java.util.Objects;
+
+// TODO(sumavg): Remove the `hide` and scope annotations.
+/** @hide */
+@RestrictTo(RestrictTo.Scope.LIBRARY)
+public abstract class AggregateField {
+  @Nullable private final FieldPath fieldPath;
+
+  @NonNull private final String operator;
+
+  @NonNull private final String alias;
+
+  private AggregateField(@Nullable FieldPath fieldPath, @NonNull String operator) {
+    this.fieldPath = fieldPath;
+    this.operator = operator;
+    // Use $operator_$field format if it's an aggregation of a specific field. For example: sum_foo.
+    // Use $operator format if there's no field. For example: count.
+    this.alias = operator + (fieldPath == null ? "" : "_" + fieldPath);
+  }
+
+  /**
+   * Returns the field on which the aggregation takes place. Returns an empty string if there's no
+   * field (e.g. for count).
+   */
+  @NonNull
+  public String getFieldPath() {
```

Comment:
`@RestrictTo(RestrictTo.Scope.LIBRARY)`

Reply:
I think `@RestrictTo` on the class is only during development. You will remove it from the class at release?

In the API proposal, we did not include a public member `getFieldPath()`. If you feel this should be included in the public API, I can make a change request.

Reply:
Sorry I see what you mean here. I will add @RestrictTo(RestrictTo.Scope.LIBRARY) to all `getFieldPath`, `getAlias`, `getOperator`.

Reply:
can you instead make it package-private? (i.e. remove the `public` keyword, but don't add `private` keyword).

Reply:
`RemoteSerializer.java` is in `package com.google.firebase.firestore.remote;` instead of `package com.google.firebase.firestore;`. So I make `getAlias()` `public` + `@RestrictTo(RestrictTo.Scope.LIBRARY)`, the rest two functions `package-private`.

Reply:
It turns out all these functions need to be used in `package com.google.firebase.firestore.remote;`, so I can only make them library wide public.